smac/optimizer/smbo.py

```diff
@@ -197,19 +197,22 @@ def run(self):
                            logger=self.logger)
 
             start_time = time.time()
-            X, Y = self.rh2EPM.transform(self.runhistory)
 
-            self.logger.debug("Search for next configuration")
-            # get all found configurations sorted according to acq
-            challengers = self.choose_next(X, Y)
+            # sample next configuration for intensification
+            challenger = self.intensifier.next_challenger(
+                challengers=None,
+                chooser=self,
+                run_history=self.runhistory,
+                repeat_configs=self.intensifier.repeat_configs
+            )
 
             time_spent = time.time() - start_time
             time_left = self._get_timebound_for_intensification(time_spent)
```

Comment:
I believe you only need to change this if SMAC actually spends time in `next_challenger`, right?

Reply:
Yes. The EPM and the local search takes a non-negligible time, so this is included after sampling the next configuration.

Reply:
Thanks for the change, however, I believe this won't work together with the current implementation in `intensification.py`. Currently, the time against which this is compared is reset to 0 in every call to `eval_challenger` in line 178. The reset to zero should probably happen after the check whether this iteration of successive halving is done. Do we have a unit test for this?